static/docs/commands-reference/push.md

```diff
@@ -1,42 +1,115 @@
-      -h, --help            show this help message and exit
-      -q, --quiet           Be quiet.
-      -v, --verbose         Be verbose.
-      -j JOBS, --jobs JOBS  Number of jobs to run simultaneously.
-      --show-checksums      Show checksums instead of file names.
-      -r REMOTE, --remote REMOTE
-                            Remote repository to push to
-      -a, --all-branches    Push cache for all branches.
-      -T, --all-tags        Push cache for all tags.
-      -d, --with-deps       Push cache for all dependencies of the specified
-                            target.
-      -R RECURSIVE, --recursive RECURSIVE
-                            Push cache from subdirectories of specified directory.
+## Options
 
-```
+* `--show-checksums` - shows checksums instead of file names.
+
+* `-r REMOTE`, `--remote REMOTE` specifies which remote cache
+  (see `dvc remote list`) to push to. The value for `REMOTE` is a cache name
+  defined using the `dvc remote` command. If no `REMOTE` is given, or if no
+  remote's are defined in the workspace, an error message is printed. If the
+  option is not specified, then the default remote, configured with the
+  `core.config` config option, is used.
+
+* `-a`, `--all-branches` - determines the files to upload by examining files
+  associated with all branches of the DVC files in the project directory. It's
+  useful if branches are used to track "checkpoints" of an experiment or
+  project.
+
+* `-T`, `--all-tags` - the same as `-a`, `--all-branches` but tags are used to
+  save different experiments or project checkpoints.
+
+* `-d`, `--with-deps` - determines the files to upload by searching backwards
+  in the pipeline from the named stage(s). The only files which will be
+  considered are associated with the named stage, and the stages which execute
+  earlier in the pipeline.
+
+* `-R`, `--recursive` -  the `targets` value is expected to be a directory path.
+  With this option, `dvc pull` determines the files to upload by searching the
+  named directory, and its subdirectories, for DVC files for which to upload
+  data. Along with providing a `target`, or `target` along with `--with-deps`,
+  it is yet another way to limit the scope of DVC files to upload.
+
+* `-j JOBS`, `--jobs JOBS` - specifies number of jobs to run simultaneously
+  while uploading files to the remote cache. The effect is to control the
+  number of files uploaded simultaneously. Default is `4 * cpu_count()`. For
+  example with `-j 1` DVC uploads one file at a time, with `-j 2` it uploads
+  two at a time, and so forth. For SSH remotes default is set to 4.
+
+* `-h`, `--help` - shows the help message and exit.
+
+* `-q`, `--quiet` - does not write anything to standard output. Exit with 0 if
+  no problems arise, otherwise 1.
+
+* `-v`, `--verbose` - displays detailed tracing information from executing the
+  `dvc push` command.
 
 ## Examples
 
+Using the `dvc push` command remote storage must be defined. For an existing
```

Comment:
See the comment above ^^. I think it worth explaining and illustrating it with more details to show state (with `tree .`) before/after, show DVC file content, show that a referenced file is in cache or not in cache, etc. It's definitely worth explaining at least at one of those example.

Reply:
I have put together an example focusing explicitly on what happens in the cache from `dvc push` operations.